server/src/main/java/org/opensearch/search/backpressure/stats/SearchBackpressureStats.java

```diff
@@ -22,21 +23,34 @@
  * Stats related to search backpressure.
  */
 public class SearchBackpressureStats implements ToXContentFragment, Writeable {
+    private final SearchTaskStats searchTaskStats;
     private final SearchShardTaskStats searchShardTaskStats;
     private final SearchBackpressureMode mode;
 
-    public SearchBackpressureStats(SearchShardTaskStats searchShardTaskStats, SearchBackpressureMode mode) {
+    public SearchBackpressureStats(
+        SearchTaskStats searchTaskStats,
+        SearchShardTaskStats searchShardTaskStats,
+        SearchBackpressureMode mode
+    ) {
+        this.searchTaskStats = searchTaskStats;
         this.searchShardTaskStats = searchShardTaskStats;
         this.mode = mode;
     }
 
     public SearchBackpressureStats(StreamInput in) throws IOException {
-        this(new SearchShardTaskStats(in), SearchBackpressureMode.fromName(in.readString()));
+        searchShardTaskStats = new SearchShardTaskStats(in);
+        mode = SearchBackpressureMode.fromName(in.readString());
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            searchTaskStats = in.readOptionalWriteable(SearchTaskStats::new);
+        } else {
+            searchTaskStats = null;
+        }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.startObject("search_backpressure")
+            .field("search_task", searchTaskStats)
```

Comment:
You probably don't want to add `search_task` if searchTaskStats is `null` (because it could be)

Reply:
Updated in the next commit.